source/mir/stat/distribution/log_normal.d

```diff
@@ -0,0 +1,306 @@
+/++
+This module contains algorithms for the Log-normal probability distribution.
+
+License: $(HTTP www.apache.org/licenses/LICENSE-2.0, Apache-2.0)
+
+Authors: John Michael Hall
+
+Copyright: 2022 Mir Stat Authors.
+
++/
+
+module mir.stat.distribution.log_normal;
+
+import mir.internal.utility: isFloatingPoint;
+
+
+/++
+Computes the Log-normal probability distribution function (PDF).
+
+Params:
+    x = value to evaluate PDF
+
+See_also:
+    $(LINK2 https://en.wikipedia.org/wiki/Log-normal_distribution, Log-normal distribution)
++/
+@safe pure nothrow @nogc
+T logNormalPDF(T)(const T x)
+    if (isFloatingPoint!T)
+    in (x >= 0, "x must be greater than or equal to zero")
+{
+    import mir.math.common: log;
+    import mir.stat.distribution.normal: normalPDF;
+    if (x == 0) return 0;
+    return x.log.normalPDF / x;
```

Comment:
Does`double.min_normal * double.epsilon` returns valid result

Reply:
Not `0`, `double.min_normal * double.epsilon >= 0`.

Btw, if the limit exists (incident infinity), the has to handle it.

Reply:
I think I see what you're saying, though not 100%. Are you saying that the contract should be `in(x >= T.min_normal * T.epsilon)` rather than `x > 0`. Anything positive value smaller than this bound should not be representable (at least for doubles). For instance, replacing `x=1e-324` in the code below gives an error. Or, are you saying that I should keep the contract what it was before and adjust the if statement to be something like `if (x < T.min_normal * T.epsilon) return 0;`? 

```d
import mir.math.func.normal;
import mir.math.common: log;
import std.stdio: writeln;

void main() {
    double x = double.min_normal * double.epsilon;
    double y = normalPDF(log(x)) / x;
    writeln(x);
    writeln(y); //prints 0
}
```

Reply:
The `x >= 0` are good. I mean we have to have tests for 0, for `T.min_normal * T.epsilon` (and for `T.min_normal` too)

Reply:
Thanks. Should be fixed.